ccvm_simulators/post_processor/adam.py

```diff
@@ -12,14 +12,27 @@ def __init__(self):
         self.pp_time = 0
         self.method_type = MethodType.Adam
 
-    def postprocess(self, c, q_matrix, v_vector, num_iter=1, device="cpu"):
+    def postprocess(
+        self,
+        c,
+        q_matrix,
+        v_vector,
+        lower_clamp=0.0,
+        upper_clamp=1.0,
+        num_iter=1,
+        device="cpu",
+    ):
         """Post processing using Adam method.
 
         Args:
             c (torch.tensor): The values for each
             variable of the problem in the solution found by the solver.
             q_matrix (torch.tensor): The Q matrix describing the BoxQP problem.
             v_vector (torch.tensor): The V vector describing the BoxQP problem.
+            lower_clamp (float, optional): Lower bound of the box constraints. Defaults
+                to 0.0.
+            upper_clamp (float, optional): Upper bound of the box constraints. Defaults
```

Comment:
I'm addressing the comment Farhad made. https://github.com/1QB-Information-Technologies/ccvm/pull/141#discussion_r1448027749

Since it's not a big change, I thought it would be convenient to incorporate it within this PR.

Reply:
Okay, I had it confused with this issue:
https://github.com/1QB-Information-Technologies/ccvm/issues/70

In general, let's be careful not to bloat our PRs too much; it can delay the review process and make it harder to find where changes were implemented. 
I like that you added these changes to the PR description, can we make sure that the PR's name represents the content, too? If not, we should do the change in a separate PR so it's easier to track/refer back to.

Reply:
Thanks for renaming 👍

Reply:
This PR now includes the implementation and tests for gradient descent, along with changes to the overall test structure. Unrelated changes have been relocated to separate PRs.

- https://github.com/1QB-Information-Technologies/ccvm/pull/150
- https://github.com/1QB-Information-Technologies/ccvm/pull/152